app/forms/steps/client/details_form.rb

```diff
@@ -24,15 +24,28 @@ def date_of_birth_ten_or_more_years_ago
         errors.add(:date_of_birth, :client_under_ten) if date_of_birth.beginning_of_day > 10.years.ago
       end
 
-      def persist!
+      def persist! # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
         return true unless changed?
 
-        applicant.update(
-          attributes.merge(
-            # The following are dependent attributes that need to be reset
-            attributes_to_reset
+        ::CrimeApplication.transaction do
+          # CRIMAPP-1313 prevents resubmitted application date_of_birth changes to
+          # maintain original application means passporting
+          if crime_application.resubmission? && changed?(:date_of_birth)
+            if applicant.over_18_at_date_stamp? && Applicant.under_18?(Time.zone.now, date_of_birth)
+              crime_application.update!(is_means_tested: 'yes')
+              applicant.update!(confirm_dwp_result: nil)
+            elsif applicant.under_18_at_date_stamp? && Applicant.over_18?(Time.zone.now, date_of_birth)
+              applicant.update!(confirm_dwp_result: nil)
+              crime_application.update!(is_means_tested: nil)
```

Comment:
I think the logic is correct, but the attributes updated are not what we'd want to update. Give me a sec to figure out which I think it would be

Reply:
So:
`is_means_tested` we want to leave alone for both (this is the Is the application means tested Q)

If they were under 18 at submission, they will not have seen:
NINO or any of the NINO / DWP confimation pages, so they will be nil and do not need resetting
I wonder if the rest of what we would need then would be covered by running the means_passporter 'Passporting::MeansPassporter.new(crime_application).call'

Bc at the moment, the validator knows something is wrong, but the decision tree isn't routing the user to the pages that they need to fulfil - and that's bc the application still has the passport in the passport array. 

I _think_ that going the other way (over 18 to under 18) that the passporter will be run and populate the initially empty array with the accurate passport. And I think that then handles the routing being correct.

What hasn't happened that I expected to is that the serialiser hasn't taken out the partner qs playback on the CYA.

Screenshot below, left is resubmission going from over18 to under18, right is initial sub uder18
![image](https://github.com/user-attachments/assets/d1d05f2b-eecd-4697-9136-fcd5735afc9d)